mouseion/mouseion/tasks.py

```diff
@@ -140,51 +174,76 @@ def process_tasks(self) -> None:
+            while self.running_tasks:
+                for task_id, process, progress_queue in self.running_tasks:
                     while not progress_queue.empty():
                         progress_value = progress_queue.get()
-                        progress.update(task_progress_map[task_id], advance=progress_value)
+                        progress.update(
+                            task_progress_map[task_id], advance=progress_value
+                        )
                     if not process.is_alive():
                         self.completed_tasks.add(task_id)
                         progress.update(task_progress_map[task_id], completed=True)
 
-                # Clean up completed processes
-                processes = [(task_id, p, pq) for task_id, p, pq in processes if p.is_alive()]
-
-    def get_task_size(self, task_id: int) -> int:
-        for task in self.all_tasks:
-            if task.id == task_id:
-                return task.size
-        return 0
+                self.running_tasks = [
+                    (task_id, p, pq)
+                    for task_id, p, pq in self.running_tasks
+                    if p.is_alive()
+                ]
 
-    def task_wrapper(self, task_id: int, function: Callable[int, Queue], size: int, progress_queue: Queue):
-        result = function(size, progress_queue)
+    def task_wrapper(
+        self,
+        task_id: int,
+        function: Callable[..., list],
+        size: int,
+        progress_queue: Queue,
+        **kwargs,
+    ):
+        result = function(size, progress_queue, **kwargs)
         for task in self.all_tasks:
             if task.id == task_id:
-                for r in result:
-                    task.result.append(r)
+                task.result.extend(result)
 
     def get_task_result(self, task_id: int) -> List:
         for task in self.all_tasks:
             if task.id == task_id:
                 return task.result
 
+
 def task_load_jsonl(size: int, progress_queue: Queue) -> list:
     progress = 0
     for i in range(size):
         time.sleep(0.3)
         progress += 1
         progress_queue.put(1)
-    return ['testc', 'testd']
+    return ["testc", "testd"]
+
 
 def task_tokenize(size: int, progress_queue: Queue) -> list:
     progress = 0
     for i in range(size):
         time.sleep(0.3)
         progress += 1
+
+    def get_task_result(self, task_id: int) -> List:
+        for task in self.all_tasks:
+            if task.id == task_id:
+                return task.result
+
+
+def task_load_jsonl(size: int, progress_queue: Queue, **kwargs) -> list:
+    for i in range(size):
+        time.sleep(0.3)
+        progress_queue.put(1)
+    return ["testc", "testd"]
+
+
+def task_tokenize(size: int, progress_queue: Queue, **kwargs) -> list:
+    for i in range(size):
+        time.sleep(0.3)
         progress_queue.put(1)
-    return ['testa', 'testb']
+    return ["testa", "testb"]
```

Comment:
as with the previous comment, returning a static value seems perhaps unintentional...